test/functional/p2p_blocksonly.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test p2p blocksonly"""
+
+from test_framework.messages import msg_tx, CTransaction, FromHex
+from test_framework.mininode import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class P2PBlocksOnly(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+        self.extra_args = [["-blocksonly"]]
+
+    def run_test(self):
+        self.nodes[0].add_p2p_connection(P2PInterface())
+
+        self.log.info('Check that txs from p2p are rejected')
+        prevtx = self.nodes[0].getblock(self.nodes[0].getblockhash(1), 2)['tx'][0]
+        rawtx = self.nodes[0].createrawtransaction(
+            inputs=[{
+                'txid': prevtx['txid'],
+                'vout': 0
+            }],
+            outputs=[{
+                self.nodes[0].get_deterministic_priv_key().address: 50 - 0.00125
+            }],
+        )
+        sigtx = self.nodes[0].signrawtransactionwithkey(
+            hexstring=rawtx,
+            privkeys=[self.nodes[0].get_deterministic_priv_key().key],
+            prevtxs=[{
+                'txid': prevtx['txid'],
+                'vout': 0,
+                'scriptPubKey': prevtx['vout'][0]['scriptPubKey']['hex'],
+            }],
+        )['hex']
+        assert_equal(self.nodes[0].getnetworkinfo()['localrelay'], False)
+        with self.nodes[0].assert_debug_log(['transaction sent in violation of protocol peer=0']):
+            self.nodes[0].p2p.send_message(msg_tx(FromHex(CTransaction(), sigtx)))
+            self.nodes[0].p2p.sync_with_ping()
```

Comment:
Maybe I'm just being thick here, but shouldn't we assert that the mininode hasn't received a message containing the new tx from `self.nodes[0]` after this `sync_with_ping()`?

Reply:
You mean spinning up a second mininode to check that that one didn't receive it?

I think adding a getmempoolinfo()['size']==0 should do the same?

Reply:
Yep, the latter sounds good.